### PR TITLE
Added MvxBindings to Forms integration

### PR DESCRIPTION
**Pull request by LeRondPoint**
Similar to the Windows implementations, we use the BindableProperties to hook up the Mvx binding engine in Forms

### CODE REVIEW

**Comment by Cheesebaron** [comment]
Nice. Great first time contribution 🙌

**Comment by LeRondPoint** [comment]
Thanks, will look into the La.ng and then I think it's possible to add a markup extension to make thin more inline with the regular bindings, allowing for something like
<,.. Text="{MvxBinding blahblah}" />

**Comment by martijn00** [comment]
Nice work! I'm thinking we should rework the structure of the Forms projects a bit like the other MvvmCross projects are. We have 1 PCL project containing all shared stuff and then the other platform specific projects added to that. Currently we have only projects which contain the presenter, but in the new structure we can add the presenter stuff in a folder, and the bindings in another. This will then also allow to develop other Forms features into the project.

**Comment by LeRondPoint** [comment]
Yes Martin I agree, event the namespaces aren't standardized in the Forms project.  In fact, shouldn't Forms be handled just like a different platform but in it's own Nuget?

**Comment by martijn00** [comment]
Yeah i think we should do that. Let me review this PR, and merge it asap. After that i'll fix up the projects and nuget.

File: MvvmCross/Droid/Droid/Platform/MvxAndroidSetup.cs [hunk 1]
```diff
@@ -158,7 +159,7 @@ protected virtual void RegisterBindingBuilderCallbacks()
             Mvx.CallbackWhenRegistered<IMvxNamespaceListViewTypeResolver>(FillNamespaceListViewTypeResolver);
         }
 
-        protected virtual MvxAndroidBindingBuilder CreateBindingBuilder()
+        protected virtual MvxBindingBuilder CreateBindingBuilder()
```
Comment (martijn00) [inline MvvmCross/Droid/Droid/Platform/MvxAndroidSetup.cs]: Why does this need to be changed?

Comment (LeRondPoint) [inline MvvmCross/Droid/Droid/Platform/MvxAndroidSetup.cs]: So we can use the overloaded class without reimplementing the InitializeBindingBuilder.  Look at the Android Setup.cs

Comment (martijn00) [inline MvvmCross/Droid/Droid/Platform/MvxAndroidSetup.cs]: Aah right, and it should still be compatible.

Comment (LeRondPoint) [inline MvvmCross/Droid/Droid/Platform/MvxAndroidSetup.cs]: Absolutely

**Comment by martijn00** [comment]
I'll rework the structure a bit now. Are you on Slack by any chance?

File: MvvmCross/Droid/Droid/Platform/MvxAndroidSetup.cs (hunk 1 above)
Comment (LeRondPoint) [inline MvvmCross/Droid/Droid/Platform/MvxAndroidSetup.cs]: This leads me to a question...  Should we have a MvxFormsSetup base class?  Not sure if it's possible without reqriting too much code...

**Comment by LeRondPoint** [comment]
Only on my Toptal account  sylvain@toptal.com

File: MvvmCross/Droid/Droid/Platform/MvxAndroidSetup.cs (hunk 1 above)
Comment (martijn00) [inline MvvmCross/Droid/Droid/Platform/MvxAndroidSetup.cs]: I was already thinking that we should have that, and possible even more base classes. The problem for me is that i don't have too many real world Forms apps where i could test it. We would really appreciate any contributions on this.

**Comment by martijn00** [comment]
Can you join the MvvmCross channel on https://xamarinchat.herokuapp.com/

**Pull request merged**